src/Control/Carrier/Class.hs

```diff
@@ -36,6 +43,19 @@ class (HFunctor sig, Monad m) => Carrier sig m | m -> sig where
   -- | Construct a value in the carrier for an effect signature (typically a sum of a handled effect and any remaining effects).
   eff :: sig m a -> m a
 
+-- | @m@ is a carrier for @sig@ containing @eff@.
+--
+-- Note that if @eff@ is a sum, it will be decomposed into multiple 'Member' constraints. While this technically allows one to combine multiple unrelated effects into a single 'Has' constraint, doing so has two significant drawbacks:
+--
+-- 1. Due to [a problem with recursive type families](https://gitlab.haskell.org/ghc/ghc/issues/8095), this can lead to significantly slower compiles.
+--
+-- 2. It defeats @ghc@’s warnings for redundant constraints, and thus can lead to a proliferation of redundant constraints as code is changed.
+type Has eff sig m = (Members eff sig, Carrier sig m)
+
+-- | Construct a request for an effect to be interpreted by some handler later on.
+send :: (Member eff sig, Carrier sig m) => eff m a -> m a
+send = eff . inj
+{-# INLINE send #-}
 
```

Comment:
Could you please add another line of whitespace here to preserve the two-line gap before the next section? 🙏